src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -246,7 +246,7 @@ protected static Point getPointOnMap(final Point point, final int xOffset, final
     final int y = point.y;
     final int width = dimension.width;
     final int screenWidth = screenDimension.width;
-    if (x - width > xOffset || (xOffset < 0 && xOffset + screenWidth < x)) {
+    if (x - width > xOffset || (xOffset < - width / 2 && xOffset + screenWidth < x)) {
```

Comment:
could be of some help to push this boolean logic to variables or helper methods, allows for something to label 'what' the values represent, and gives us some context as to why we are doing the operations. As is, you somewhat have to reverse engineer what is going on. Ideally we would save on that effort via self-documenting code (failing that, throw some comments in)